src/ngx_http_lua_output.c

```diff
@@ -146,8 +147,23 @@ ngx_http_lua_ngx_echo(lua_State *L, unsigned newline)
         size += sizeof("\n") - 1;
     }
 
+    uri_changes = r->uri_changes;
+
     if (size == 0) {
         rc = ngx_http_lua_send_header_if_needed(r, ctx);
+
+        if (ctx != ngx_http_get_module_ctx(r, ngx_http_lua_module)
+            && r->filter_finalize)
+        {
+            if (uri_changes > r->uri_changes) {
+                ctx->filter_finalize = 1;
+                return lua_yield(L, 0);
+
+            } else {
+                ngx_http_set_ctx(r, ctx, ngx_http_lua_module);
```

Comment:
Overriding the new ctx looks too dangerous to me. When can this actually happen? Will you add a corresponding test case to actually cover this code path?

Reply:
When `r->filter_finalize` == 1, all the http modules' ctx was cleared(see `ngx_http_filter_finalize_request`), so this `ngx_http_set_ctx(r, ctx, ngx_http_lua_module);` is used to resume the ctx of ngx_lua.

I will add a corresponding test case for this.